Cookbook/Proposals/AdditionalRulesForDanger.md

```diff
@@ -0,0 +1,133 @@
+
+`\.signal\(for: \#selector\(UIViewController\.`
+
+We'd have to restrict this rule to Builders and FlowControllers only, therefore we'd have to whitelist filenames that matched `.*Builder.swift` and `.*FlowController.swift`.
+
+This rule could be enforced with either tool but I'd recommend using SwiftLint for this since due to the low number of warnings it would trigger in the current state of our database. Also, I believe we should try to use Danger for particularly important warnings in order not to spam the pull request with comments.
+
+
+### Deprecated calls of DesignLibrary components (aka pre-NVL components)
+
+Given that NVL is scheduled for release within two weeks, using NVL components is pivotal to Babylon's overall look and feel.
+With the exception of the empty space component (even though it has its own NVL equivalent), every other pre-NVL component used inside Renderer should trigger a warning.
+
+The following regex currently yields 1333 results in 216 files: `Component\.(?!EmptySpace)\b`.
+The sheer number of warnings immediately excludes SwiftLint; Danger would be our only option.
+
+Nevertheless, this would require that everyone relied exclusively on NVL components and I am not sure this is the case right now.
+
+
+### Mutating Current in `setUp()` but not restoring it in `tearDown()`
+
+The rule is simple: every object (susceptible to global mutation) that is mutated in `setUp()` must be restored in `tearDown()`.
+Mutations in `Current` in particular are the prime suspects of a number of flaky tests that have been recently affecting our test suite.
+
+In order to detect this problem we have to use Danger since SwiftLint only accepts regular expressions.
+We would have to whitelist every test file (`.*Tests.*`) and use Danger to manually parse the test line-by-line until the `setUp()` function was found. 
+
+If found, we need to make a counter that increases by one each time a curly bracket and does the reverse when a closing curly bracket appears. Once the counter reaches zero again, then we've successfully delimited the contents of the `setUp()` function and we'd use the following regex to detect changes in Current: `Current\..*\ = `.
+
+The algorithm is then be executed once more but for the `tearDown()` method instead.
+If the number of changes inside each function matches, then no issue would be found. If not, then Current appeared to have been mutated but not restored; we'd issue a warning for the lines in which the algorithm detected a change and alert the developers in GitHub.
+
+Nevertheless, this rule is rather naïve because we have no actual way of detecting if Current was reset or not; that would require compiling and runniing the actual Swift code. Therefore, this approach is not a silver bullet or anything remotely similar but it would still be useful for one use case in particular: should the developer forget to restore Current, this rule will alert us of this situation.
```

Comment:
```suggestion
Nevertheless, this rule is rather naïve because we have no actual way of detecting if Current was reset or not; that would require compiling and running the actual Swift code. Therefore, this approach is not a silver bullet or anything remotely similar but it would still be useful for one use case in particular: should the developer forget to restore Current, this rule will alert us of this situation.
```